amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -69,7 +71,30 @@ public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        if (configuration.getStreamArn() != null && !configuration.getStreamArn().trim().isEmpty()) {
+            Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
+            if(!streamArnObj.getResource().getResourceType().equalsIgnoreCase("stream")){
+                throw new IllegalArgumentException(String.format("StreamArn has unsupported resource type of '%s'. Expected: stream",
+                        streamArnObj.getResource().getResourceType()));
```

Comment:
Nit: `streamArnObj.getResource().getResourceType()` is invoked three times. This could be stashed in a local variable to simplify the code (and, subjectively, make it more legible and Easy To Change).